browser/ui/tabs/brave_tab_strip_model.cc

```diff
@@ -0,0 +1,146 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <algorithm>
+#include <memory>
+
+#include "brave/browser/ui/tabs/brave_tab_strip_model.h"
+
+#include "brave/common/pref_names.h"
+#include "build/build_config.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/web_contents.h"
+#include "ui/events/event.h"
```

Comment:
it looks like you're missing deps for some of this stuff in browser/ui/BUILD.gn
I would suggest creating a new BUILD.gn in browser/ui/tabs so we can keep the deps organized a little better. brave_tab_menu_model.cc/brave_tab_menu_model.h should also be included in that file along with any deps required for them. Don't forget to add conditional deps for `use_aura` and `is_mac` in gn

Reply:
Added a new BUILD.gn in browser/ui/tabs. I am not sure what should be added in the deps. Could you point me to some docs/guidelines of what should be there ? Currently, I added the minimum for it to compile.